### PR TITLE
- Create Edit Transaction UI

### PR DESCRIPTION
**Pull request by danacealfonso**
# 📲 What #

* Transaction edit UI

# ✅ Acceptance criteria #

- [x ] Tested it locally?

# ⏰ TODO #

* Create transaction edit UI

# 🔗 References #

Screen Shot 2021-05-17 at 5 58 03 PM

### CODE REVIEW

File: app/src/main/res/layout/fragment_transaction_add_1.xml [hunk 1]
```diff
@@ -95,6 +95,7 @@
                     android:layout_gravity="center_horizontal"
                     android:layout_marginTop="18dp"
                     android:text="₱"
+                    android:textColor="#414141"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_add_1.xml]: pls add to colors.xml

File: app/src/main/res/layout/fragment_transaction_add_1.xml [hunk 2]
```diff
@@ -109,6 +110,7 @@
                     android:layout_gravity="center_horizontal"
                     android:background="@android:color/transparent"
                     android:text="00.00"
+                    android:textColor="#414141"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_add_1.xml]: pls add to colors.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+        <TextView
+            android:id="@+id/tv_title"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="10dp"
+            android:layout_marginEnd="10dp"
+            android:ellipsize="end"
+            android:gravity="center"
+            android:singleLine="true"
+            android:text="Edit Transaction"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to strings.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+            app:layout_constraintTop_toTopOf="@+id/btn_back" />
+
+        <com.google.android.material.textview.MaterialTextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="20dp"
+            android:layout_marginEnd="20dp"
+            android:clickable="true"
+            android:elevation="30dp"
+            android:textColor="#7AF1F3F6"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to colors.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 5]
```diff
@@ -0,0 +1,421 @@
...
+                        app:layout_constraintStart_toStartOf="parent"
+                        app:layout_constraintTop_toTopOf="parent"
+                        app:layout_constraintVertical_bias="0.383">
+
+                        <TextView
+                            android:layout_width="wrap_content"
+                            android:layout_height="wrap_content"
+                            android:layout_marginStart="20sp"
+                            android:text="@string/lbl_which_wallet_do_you_want_to_add"
+                            android:textColor="#BBC3C9"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to colors.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 6]
```diff
@@ -0,0 +1,421 @@
...
+                            android:layout_marginLeft="20sp"
+                            android:layout_marginTop="25sp"
+                            android:layout_marginRight="20sp"
+                            android:layout_marginBottom="20sp"
+                            android:background="@drawable/edittext_bg_round_corner"
+                            android:elevation="10sp"
+                            android:inputType="text"
+                            android:padding="14sp"
+                            android:text="BPI Savings"
+                            android:textColor="#414141"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to colors.xml and strings.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 7]
```diff
@@ -0,0 +1,421 @@
...
+                            android:id="@+id/ib_trans_wallet"
+                            android:layout_width="40sp"
+                            android:layout_height="40sp"
+                            android:layout_alignParentEnd="true"
+                            android:layout_marginTop="32sp"
+                            android:layout_marginEnd="30sp"
+                            android:background="@drawable/roundsemi_gray"
+                            android:elevation="11sp"
+                            android:src="@drawable/ic_arrow_down_gray_24"
+                            android:text="Button"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to strings.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 8]
```diff
@@ -0,0 +1,421 @@
...
+                            tools:ignore="HardcodedText,ContentDescription" />
+                    </RelativeLayout>
+
+                    <TextView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_marginStart="20sp"
+                        android:text="Transaction amount"
+                        android:textColor="#BBC3C9"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to colors.xml and strings.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 9]
```diff
@@ -0,0 +1,421 @@
...
+                        android:layout_marginLeft="20sp"
+                        android:layout_marginTop="10sp"
+                        android:layout_marginRight="20sp"
+                        android:layout_marginBottom="20sp"
+                        android:background="@drawable/edittext_bg_round_corner"
+                        android:elevation="10sp"
+                        android:inputType="text"
+                        android:padding="14sp"
+                        android:text="₱10,000.00"
+                        android:textColor="#414141"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to colors.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 10]
```diff
@@ -0,0 +1,421 @@
...
+                            android:layout_marginLeft="20sp"
+                            android:layout_marginTop="25sp"
+                            android:layout_marginRight="20sp"
+                            android:layout_marginBottom="20sp"
+                            android:background="@drawable/edittext_bg_round_corner"
+                            android:elevation="10sp"
+                            android:inputType="text"
+                            android:padding="14sp"
+                            android:text="Income"
+                            android:textColor="#414141"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to colors.xml and strings.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 11]
```diff
@@ -0,0 +1,421 @@
...
+                            android:id="@+id/ib_trans_type"
+                            android:layout_width="40sp"
+                            android:layout_height="40sp"
+                            android:layout_alignParentEnd="true"
+                            android:layout_marginTop="32sp"
+                            android:layout_marginEnd="30sp"
+                            android:background="@drawable/roundsemi_gray"
+                            android:elevation="11sp"
+                            android:src="@drawable/ic_arrow_down_gray_24"
+                            android:text="Button"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to strings.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 12]
```diff
@@ -0,0 +1,421 @@
...
+                        app:layout_constraintStart_toStartOf="parent"
+                        app:layout_constraintTop_toTopOf="parent"
+                        app:layout_constraintVertical_bias="0.383">
+
+                        <TextView
+                            android:layout_width="wrap_content"
+                            android:layout_height="wrap_content"
+                            android:layout_marginStart="20sp"
+                            android:text="Category"
+                            android:textColor="#BBC3C9"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to colors.xml and strings.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 13]
```diff
@@ -0,0 +1,421 @@
...
+                            android:layout_width="match_parent"
+                            android:layout_height="55sp"
+                            android:layout_marginLeft="20sp"
+                            android:layout_marginTop="25sp"
+                            android:layout_marginRight="20sp"
+                            android:layout_marginBottom="20sp"
+                            android:background="@drawable/edittext_bg_round_corner"
+                            android:elevation="10sp"
+                            android:text="Salary"
+                            android:inputType="text"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to colors.xml and strings.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 14]
```diff
@@ -0,0 +1,421 @@
...
+                            android:id="@+id/ib_category"
+                            android:layout_width="40sp"
+                            android:layout_height="40sp"
+                            android:layout_alignParentEnd="true"
+                            android:layout_marginTop="32sp"
+                            android:layout_marginEnd="30sp"
+                            android:background="@drawable/roundsemi_gray"
+                            android:elevation="11sp"
+                            android:src="@drawable/ic_arrow_down_gray_24"
+                            android:text="Button"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to strings.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 15]
```diff
@@ -0,0 +1,421 @@
...
+                            tools:ignore="HardcodedText,ContentDescription"
+
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: remove space

tools:ignore="HardcodedText,ContentDescription" />

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 16]
```diff
@@ -0,0 +1,421 @@
...
+                        app:layout_constraintStart_toStartOf="parent"
+                        app:layout_constraintTop_toTopOf="parent"
+                        app:layout_constraintVertical_bias="0.383">
+
+                        <TextView
+                            android:layout_width="wrap_content"
+                            android:layout_height="wrap_content"
+                            android:layout_marginStart="20sp"
+                            android:text="Select date"
+                            android:textColor="#BBC3C9"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to colors.xml and strings.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 17]
```diff
@@ -0,0 +1,421 @@
...
+                            android:layout_marginLeft="20sp"
+                            android:layout_marginTop="25sp"
+                            android:layout_marginRight="20sp"
+                            android:layout_marginBottom="20sp"
+                            android:background="@drawable/edittext_bg_round_corner"
+                            android:elevation="10sp"
+                            android:inputType="text"
+                            android:padding="14sp"
+                            android:text="15 / 02 / 2021"
+                            android:textColor="#414141"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to colors.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 18]
```diff
@@ -0,0 +1,421 @@
...
+                            android:id="@+id/ib_trans_date"
+                            android:layout_width="40sp"
+                            android:layout_height="40sp"
+                            android:layout_alignParentEnd="true"
+                            android:layout_marginTop="32sp"
+                            android:layout_marginEnd="30sp"
+                            android:background="@drawable/roundsemi_gray"
+                            android:elevation="11sp"
+                            android:src="@drawable/ic_arrow_down_gray_24"
+                            android:text="Button"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to strings.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 19]
```diff
@@ -0,0 +1,421 @@
...
+
+                            />
+                    </RelativeLayout>
+
+                    <TextView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_marginStart="20sp"
+                        android:text="Spent with this person"
+                        android:textColor="#BBC3C9"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to colors.xml and strings.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 20]
```diff
@@ -0,0 +1,421 @@
...
+                        android:layout_height="55sp"
+                        android:layout_marginLeft="20sp"
+                        android:layout_marginTop="10sp"
+                        android:layout_marginRight="20sp"
+                        android:background="@drawable/edittext_bg_round_corner"
+                        android:elevation="10sp"
+                        android:text="Juan Dela Cruz"
+                        android:inputType="text"
+                        android:padding="14sp"
+                        android:textColor="#414141"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to colors.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 21]
```diff
@@ -0,0 +1,421 @@
...
+
+                    <TextView
+                        android:id="@+id/tv_trans_receipt"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_marginTop="20sp"
+                        android:layout_marginStart="20sp"
+                        android:layout_weight="1"
+                        android:text="@string/lbl_transaction_image"
+                        android:textColor="#BBC3C9"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to colors.xml

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 22]
```diff
@@ -0,0 +1,421 @@
...
+                        tools:ignore="ContentDescription" />
+
+                    <TextView
+                        android:id="@+id/tv_trans_filename"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_marginStart="20sp"
+                        android:layout_weight="1"
+                        android:text="receipt101020.jpg"
+                        android:textColor="#BBC3C9"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to colors.xm

File: app/src/main/res/layout/fragment_transaction_edit.xml [hunk 23]
```diff
@@ -0,0 +1,421 @@
...
+                        android:id="@+id/btn_browse"
+                        android:layout_width="150sp"
+                        android:layout_height="56sp"
+                        android:layout_marginStart="20sp"
+                        android:layout_marginTop="20dp"
+                        android:layout_marginBottom="50sp"
+                        android:background="@drawable/edittext_bg_round_corner_gray"
+                        android:backgroundTint="@color/gray"
+                        android:elevation="10sp"
+                        android:text="Replace"
```
Comment (rexendy) [inline app/src/main/res/layout/fragment_transaction_edit.xml]: pls add to strings.xml

File: app/src/main/res/navigation/nav_graph.xml [hunk 24]
```diff
@@ -56,6 +59,12 @@
             app:destination="@id/transaction_add_1_fragment" />
     </fragment>
 
+    <fragment
+        android:id="@+id/transaction_edit_fragment"
+        android:name="com.kwartracker.android.transactions.ui.edit.TransactionEditFragment"
+        android:label="Transaction Edit"
```
Comment (rexendy) [inline app/src/main/res/navigation/nav_graph.xml]: pls add to strings.xml

**Review by rexendy — CHANGES_REQUESTED** [review]
- once the api is implemented. pls remove all static values 
ex. name (Juan), image (received.jpg)

**Comment by danacealfonso** [comment]
> * once the api is implemented. pls remove all static values
>   ex. name (Juan), image (received.jpg)

will do Ms. @rexendy

**Review by rexendy — APPROVED** [review]

**Review by MarvinBlanca — APPROVED** [review]

**Review by jupilot11 — APPROVED** [review]

**Review by kalvinbdr — APPROVED** [review]

**Pull request merged**